snippets/core/system-text-json/csharp/SerializePolymorphic.cs

```diff
@@ -14,35 +14,80 @@ public static void Run()
             {
                 WriteIndented = true
             };
-            jsonString = JsonSerializer.Serialize<WeatherForecast>(weatherForecast, serializeOptions);
+            jsonString = JsonSerializer.Serialize<WeatherForecast>(weatherForecast, options);
             // </SnippetSerializeDefault>
 
             Console.WriteLine($"JSON output:\n{jsonString}\n");
 
             Console.WriteLine("Object generic type parameter - derived class properties included");
             // <SnippetSerializeObject>
-            serializeOptions = new JsonSerializerOptions
+            options = new JsonSerializerOptions
             {
                 WriteIndented = true
             };
-            jsonString = JsonSerializer.Serialize<object>(weatherForecast, serializeOptions);
+            jsonString = JsonSerializer.Serialize<object>(weatherForecast, options);
             // </SnippetSerializeObject>
             Console.WriteLine($"JSON output:\n{jsonString}\n");
 
 
             Console.WriteLine("GetType() type parameter - derived class properties included");
             // <SnippetSerializeGetType>
-            serializeOptions = new JsonSerializerOptions
+            options = new JsonSerializerOptions
             {
                 WriteIndented = true
             };
-            jsonString = JsonSerializer.Serialize(weatherForecast, weatherForecast.GetType(), serializeOptions);
+            jsonString = JsonSerializer.Serialize(weatherForecast, weatherForecast.GetType(), options);
             // </SnippetSerializeGetType>
             Console.WriteLine($"JSON output:\n{jsonString}\n");
+
+            Console.WriteLine("Extra properties on interface implementations included only for object properties");
+            // <SnippetSerializeInterface>
+            var forecasts = new Forecasts
+            {
+                Monday = new Forecast
+                {
+                    Date = DateTime.Parse("2020-01-06"),
+                    TemperatureCelsius = 10,
+                    Summary = "Cool",
+                    WindSpeed = 8
+                },
+                Tuesday = new Forecast
+                {
+                    Date = DateTime.Parse("2020-01-07"),
+                    TemperatureCelsius = 11,
+                    Summary = "Rainy",
+                    WindSpeed = 10
+                }
+            };
+
+            options = new JsonSerializerOptions
+            {
+                WriteIndented = true
+            };
+            Console.WriteLine(JsonSerializer.Serialize(forecasts, options));
+            // </SnippetSerializeInterface>
+
+            WeatherForecastWithPrevious weatherForecastWithPrevious = WeatherForecastFactories.CreateWeatherForecastWithPrevious();
+            WeatherForecastWithPreviousAsObject weatherForecastWithPreviousAsObject = WeatherForecastFactories.CreateWeatherForecastWithPreviousAsObject();
+
+            Console.WriteLine();
```

Comment:
Do we have such new lines elsewhere? It doesn't look like it (at least on this file). Why is it needed here? Does the output look better because of it (and if so, it makes sense to do so, but then why not elsewhere)?

Reply:
Elsewhere I kept the WriteLine calls out of the snippets and added `\n` where needed to make the output readable.  In this instance I left the WriteLine in the snippet but didn't want to append a newline to it that would be visible in the doc so I added the call to WriteLine. For consistency with the other snippets I could send the serialization output in this snippet to jsonString instead of WriteLine.

Reply:
Ah yes, you mean the previous output on [line 71](https://github.com/dotnet/samples/pull/1897/files#diff-a01210cbfe705b0745641383f463e05bR71). Makes sense to be consistent. Up to you.